lib/course_refresher.rb

```diff
@@ -129,12 +130,12 @@ def refresh_course(course)
         rescue StandardError, ScriptError # Some YAML parsers throw ScriptError on syntax errors
           @report.errors << $!.message + "\n" + $!.backtrace.join("\n")
           # Delete the new cache we were working on
-          FileUtils.rm_rf(@course.cache_path)
+          FileUtils.rm_rf(@course.cache_path)                     unless options[:no_directory_changes]
```

Comment:
I'd consider just having one space before the unless - this looks like too much indentation and adds overhead to change it if there is more steps added to the refresh process